### PR TITLE
Add verify-gofmt as a Bazel test.

### PR DESCRIPTION
**Pull request by spxtr**
Also, moved `--flaky_test_attempts=3` out of `.bazelrc` since it was annoying. I think this verify-all pattern is working well in test-infra, we should try and make it happen here.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: hack/verify-gofmt.sh [hunk 1]
```diff
@@ -23,10 +23,15 @@ set -o pipefail
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 
-kube::golang::verify_go_version
-
 cd "${KUBE_ROOT}"
 
+# Prefer bazel's gofmt.
+gofmt="external/io_bazel_rules_go_toolchain/bin/gofmt"
```
Comment (ixdy) [inline hack/verify-gofmt.sh]: this makes me a little sad, but I'm not sure if there's a better way (besides adding this to the PATH?)

File: hack/verify-gofmt.sh [hunk 2]
```diff
@@ -38,14 +43,13 @@ find_files() {
         -o -wholename '*/third_party/*' \
         -o -wholename '*/vendor/*' \
         -o -wholename './staging' \
+        -o -wholename '*/bindata.go' \
```
Comment (ixdy) [inline hack/verify-gofmt.sh]: why is this necessary?

File: hack/verify-gofmt.sh (hunk 1 above)
Comment (spxtr) [inline hack/verify-gofmt.sh]: I think rules_go can expose this more nicely along with the raw go tool, but until that happens this is about as good as we're going to do.

File: hack/verify-gofmt.sh (hunk 2 above)
Comment (spxtr) [inline hack/verify-gofmt.sh]: We're actually generating `test/e2e/generated/bindata.go`, and I guess it needs gofmt, as in [here](https://github.com/kubernetes/kubernetes/blob/master/hack/generate-bindata.sh#L53)...

**Comment by spxtr** [comment]
Manually applying approved.
